plasmapy/diagnostics/thomson.py

```diff
@@ -11,75 +11,78 @@
 from astropy import units as u
 import astropy.constants as const
 
-from plasmapy.formulary.dispersionfunction \
-    import plasma_dispersion_func_deriv as ZPrime
-
 from plasmapy.utils.decorators import validate_quantities
-from plasmapy.formulary.parameters import plasma_frequency as wp
+from plasmapy.formulary.parameters import plasma_frequency
+
+from plasmapy.formulary.dielectric import permittivity_1D_Maxwellian
+
+from plasmapy.particles import Particle
 
 
 # TODO: interface for inputting a multi-species configuration could be
 # simplified using the plasmapy.classes.plasma_base class if that class
 # included ion and electron drift velocities and information about the ion
 # atomic species.
 
-@validate_quantities(wavelength=u.nm, probe_wavelength=u.nm, ne=u.cm**-3,
-                  Te=u.eV, Ti=u.eV, ion_vel=u.cm/u.s, fluid_vel=u.cm/u.s)
-def spectral_density(wavelength, probe_wavelength=532*u.nm, ne=1e15*u.cm**-3,
-                     fract=np.ones(1), Te=1*u.eV, Ti=np.ones(1)*u.eV,
-                     ion_z=np.ones(1), ion_mu=np.ones(1),
+@validate_quantities(wavelengths=u.nm, probe_wavelength=u.nm, ne=u.cm**-3,
+                  kTe=u.eV, kTi=u.eV, ion_vel=u.cm/u.s, fluid_vel=u.cm/u.s)
+def spectral_density(wavelengths, probe_wavelength, ne, kTe, kTi,
```

Comment:
I completely forgot about the possibility of converting the input to `Particle`s! If you're doing it that way, you might find http://docs.plasmapy.org/en/latest/api/plasmapy.particles.particle_input.html helpful!

Reply:
I made a [comment](https://github.com/PlasmaPy/PlasmaPy/pull/831#discussion_r436305743) to move towards using the `Particle` class.  Unfortunately, the `ion_species` argument can be a list of of `Particles`, which `particle_input` can not handle (Correct me if I"m wrong on that).

Reply:
It's a tough little decorator, it'll handle it :)

Reply:
The issue is...`particle_input` does not recognize more detailed typing annotations.  For example, it will know a typing of just `Particle` but not `List[Particle]`.

<img width="877" alt="image" src="https://user-images.githubusercontent.com/29869348/83997400-2779fc00-a913-11ea-80d3-865a4616f994.png">